src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/IDependencyContextProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    /// <summary>
+    /// Exposes a <see cref="DependencyContext"/> from an <see cref="ApplicationPart"/>.
+    /// </summary>
+    public interface IDependencyContextProvider
```

Comment:
Can you do `CompileLibrariesProvider`, you don't need anything else from context.